bin/rust/rust_belt/lifetime_logic.gh

```diff
@@ -42,7 +45,17 @@ lemma void close_lifetime_intersection_dead_token(lifetime_t k, lifetime_t k1);
     requires [_]lifetime_dead_token(k);
     ensures [_]lifetime_dead_token(lifetime_intersection(k, k1));
 
-predicate thread_token(thread_id_t t); // Non-atomic token with Top mask ([NaInv: t.Top] in RustBelt)
+fixpoint void *Nshr() { return (void *) 0; }
+fixpoint void *Nrust() { return (void *) 1; }
+
+predicate thread_token(thread_id_t t, mask_t m);
```

Comment:
Should we add a lemma that allows one to replace the mask_t value in a thread_token by an equivalent value? This way canonicity is not crucial.